### ISSUE TITLE
Feature Request: EC2 Image Builder Distribution Configuration

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#13316]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In December 2019 AWS announced the availability of EC2 ImageBuilder, a service that makes it easier and faster to build and maintain secure images. Image Builder simplifies the creation, patching, testing, distribution, and sharing of Linux or Windows Server images. 

### New Resource

* aws_imagebuilder_distribution_configuration

### Potential Terraform Configuration

```hcl
resource "aws_imagebuilder_distribution_configuration" "example" {
  arn                = ""  
  client_token       = ""
  description        = ""
  distribution {           
    ami_distribution_configuration {
      ami_tags         = {}
      description      = ""
      launch_permission {
        user_groups      = []
        user_ids         = []
      }
      name             = ""
    }
    license_configuration_arns = []
    region                     = ""
  }
  name               = ""
  tags               = {}
  timeout_minutes    = 0 
}
```

### References

Announcement: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-ec2-image-builder/
API Reference: https://docs.aws.amazon.com/imagebuilder/latest/APIReference/Welcome.html
SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/imagebuilder/
* #11084

**Comment by iancward** [comment]
Is this a dupe of #11084? That one appears to encompass the full service.

**Comment by breathingdust** [comment]
Hi @iancward, it's not a dupe I'm just breaking it out into separate resources. It's easier for us to track that way, and easier for contributors to pick up a single resource. 👍

**Comment by iancward** [comment]
Might be resolved by #13485

**Comment by bflad** [comment]
A new `aws_imagebuilder_distribution_configuration` resource has been merged and will release with version 3.16.0 of the Terraform AWS Provider, later this week. Thank you to @Dogers for the initial implementation. 👍
